modules/deriving/src/main/scala/shapeless3/deriving/kinds.scala

```diff
@@ -315,28 +306,27 @@ object K1:
     inline def widen[G[t[_]] >: F[t]]: CoproductInstances[G, T] =
       inst.asInstanceOf
 
-  inline given mkInstances[F[_[_]], T[_]](using gen: Generic[T]): Instances[F, T] =
+  inline given mkInstances[F[_[_]], T[_]](using gen: Refl[Mirror, T]): Instances[F, T] =
```

Comment:
Isn't this exactly the same as before? `Refl` still has the `type Kind` refinement and will thus exclude regular `Mirror`-instances from the implicit search that do not have this refinement.

Reply:
No, it's not the same. `Refl` has no type member `Kind` (`Generic` has it).
```scala
  type Refl[M, O[_]] = M {
    type MirroredType[X] = O[X]
    type MirroredMonoType = O[Any]
    type MirroredElemTypes[_] <: Tuple
  }
```

Basically `Refl[Mirror, F]` is the equivalent of `Mirror.Of` for kind `F[_]`

Reply:
For comparison here is the definition of `Mirror.Of`:
```scala
type Of[T] = Mirror {
  type MirroredType = T
  type MirroredMonoType = T
  type MirroredElemTypes <: Tuple
}
```

Reply:
I see, you are right. `Generic` is then only needed for the extension methods that you mentioned?

Reply:
I believe so, yeah